app/src/main/java/ch/coredump/watertemp/activities/MapActivity.kt

```diff
@@ -555,6 +555,24 @@ class MapActivity : AppCompatActivity(), OnMapReadyCallback {
         this@MapActivity.chart3days!!.invalidate()
     }
 
+    private fun updateDataSummary(measurements: List<Measurement>) {
+        var minTemp: Float? = null
+        var maxTemp: Float? = null
+        var avgTemp: Float = 0.0f
+        for (temperature in measurements.map { it.temperature }) {
+            if (minTemp == null || minTemp > temperature) {
+                minTemp = temperature
+            } else if (maxTemp == null || maxTemp < temperature) {
+                maxTemp = temperature
+            }
+            avgTemp += temperature
+        }
```

Comment:
There's an edge case for a collection containing downward sorted temperatures. For example `[4, 3, 2, 1]` or `[24]` would leave `maxTemp` undefined. That's why I avoid `else if`.

Anyways:
I'd prefer three iterations for readability. This is not performance critical, so I'd write something like this:

```java
minTemp = measurements.stream().map { it.temperature }.average().orElse(null)
maxTemp = measurements.stream().map { it.temperature }.min().orElse(null)
avgTemp = measurements.stream().map { it.temperature }.max().orElse(null)
```

Reply:
Good catch, thanks! :slightly_smiling_face:

Reply:
Regarding `.stream()`, that requires at least API level 24 (Android 7). Current API level is 17. I see no issues with raising it to 21 (Android 5), but 24 is a bit high.

Luckily Kotlin features native sequences: https://kotlinlang.org/docs/reference/sequences.html (Alternatively, there's a backport of the streams here: https://github.com/stefan-zobel/streamsupport But since we already use Kotlin it makes sense to use sequences.)